### PR TITLE
Replace deprecated request library with simple-get

### PR DESCRIPTION
**Pull request by anoadragon453**
The `request` package [has been deprecated](https://github.com/request/request/issues/3143), and thus this PR aims to replace it with something else.

I initially tried to use [superagent](https://www.npmjs.com/package/superagent) which has a similar API to `request`, but had a lot of difficulty with both piping the response body to a file, as well as extracting the response headers. Instead, I opted for a much more lightweight library ([simple-get](https://github.com/feross/simple-get)) which got out of the way and allowed for more flexibility.

A helper method was needed in `config.js` to generate an appropriate proxy tunnel for HTTP vs. HTTPS proxy URLs.

This PR also fixed some typos and docstrings in the surrounding source code.

### CODE REVIEW

File: src/reporting.js [hunk 1]
```diff
@@ -215,39 +216,52 @@ async function _generateReportFromDownload(req, domain, mediaId, matrixFile, opt
...
-            let responseHeaders;
-	    let connect = {url: httpUrl, encoding: null, qs: thumbnailQueryParams};
-	    const config = getConfig();
-	    const proxy = config.proxy;
-	    if (proxy != null) {
-		connect.proxy = proxy;
-	    }
-	    if (config.requestHeader != null) {
-		connect.headers = generateRequestHeaders(config.requestHeader, req);
-		console.info(`Request headers are`, connect.headers);
```
Comment (anoadragon453) [inline src/reporting.js:231]: I also switched this from a `console.info` to `console.debug` as I don't think it needs to be at the info level...

This line was added recently in dae7bcd51227f32123fbd181738058fdd62ce1e9.

File: src/reporting.js [hunk 2]
```diff
@@ -215,39 +216,52 @@ async function _generateReportFromDownload(req, domain, mediaId, matrixFile, opt
     console.info(`Downloading ${httpUrl}, writing to ${filePath}`);
 
     let downloadHeaders;
-    let response;
```
Comment (anoadragon453) [inline src/reporting.js:218]: This was unused.

File: src/reporting.js [hunk 3]
```diff
@@ -215,39 +216,52 @@ async function _generateReportFromDownload(req, domain, mediaId, matrixFile, opt
...
-                })
-                .on('end', () => {
-                    resolve(responseHeaders);
-                })
-                .pipe(fileWriteStream);
+            // Base options for the request
+            const opts = {
+                url: httpUrl,
+                headers: {
+                    'content-type': 'application/octet-stream',
```
Comment (dbkr) [inline src/reporting.js:226]: Looks like this is just making get requests, so shouldn't be sending a content-type?

File: src/reporting.js [hunk 4]
```diff
@@ -215,39 +216,52 @@ async function _generateReportFromDownload(req, domain, mediaId, matrixFile, opt
...
+                }
+            };
+
+            const config = getConfig();
+
+            // Add additional request headers if configured
+            if (config.requestHeader != null) {
+                opts.headers += generateRequestHeaders(config.requestHeader, req);
```
Comment (dbkr) [inline src/reporting.js:234]: This looks like it's using `+` on two objects which I think does something very silly in js - you probably want `Object.assign()`.

**Review by dbkr — COMMENTED** [review]
Seems reasonable. Fwiw I've used needle in the past and it seemed pretty good, but simple-get seems fine for this purpose.

One small nit: the tunnel creation code could probably use a separate file rather than being lumped into the config.

File: package.json [hunk 5]
```diff
@@ -17,9 +17,9 @@
     "file-type": "^10.5.0",
     "joi": "^13.3.0",
     "js-yaml": "^3.12.0",
-    "request": "^2.87.0",
-    "request-promise": "^4.2.2",
-    "rimraf": "^2.6.2"
+    "rimraf": "^2.6.2",
+    "simple-get": "^4.0.0",
```
Comment (turt2live) [inline package.json:21]: honorary "why not node-fetch" questions apply :p

Comment (anoadragon453) [reply package.json:21]: "I don't know what the node landscape is like, but this seemed to work" response :D

Comment (turt2live) [reply package.json:21]: fair. node-fetch handles a bit more than get for when we get around to it, but simple-get is the same idea. Either is fine, really.

**Review by dbkr — APPROVED** [review]

**Pull request merged**